cpp/benchmarks/fixture/rmm_pool_raii.hpp

```diff
@@ -69,4 +69,15 @@ class rmm_pool_raii {
   std::shared_ptr<rmm::mr::device_memory_resource> mr;
 };
 
+/**
+ * Base fixture for cudf benchmarks using nvbench.
+ *
+ * Initializes the default memory resource to use the RMM pool device resource.
+ */
+struct nvbench_base_fixture {
+  rmm_pool_raii _mr;
+};
+
 }  // namespace cudf
+
+#define NVBENCH_ENVIRONMENT cudf::nvbench_base_fixture
```

Comment:
Could you use the `rmm_pool_raii` directly here? If the goal is to restrict the scope to more nvbench-specific things, the alternative question is, does `rmm_pool_raii` have any other purpose or could we just rename it to `nvbench_base_fixture`?

Reply:
As an aside, while we're making this change should we change the namespace? Our GTest fixture is in cudf::test, maybe this object should live in cudf::benchmark.

Reply:
I wanted to just wrap in case we wanted add some other specific logic. I'd assume it had some other purpose. Regardless I felt it was clearer in its desired intent in the current implementation.
There is no `cudf::benchmark` namespace. I think creating the namespace would be out of scope for this PR.